cli/commands/wp.ts

```diff
@@ -1,81 +1,113 @@
+export async function runCommand(
+	siteFolder: string,
+	args: string[],
+	options: {
+		phpVersion?: string;
+	} = {}
+): Promise< void > {
 	const site = await getSiteByFolder( siteFolder );
 
-	// If there's already a running Playground instance for this site, pass the command to it…
+	// Determine the PHP version to use
+	const phpVersionSchema = z.enum( SupportedPHPVersions );
+	let phpVersion: SupportedPHPVersion;
+	const requestedPhpVersion = options.phpVersion ?? site.phpVersion;
+
 	try {
-		await connect();
+		phpVersion = phpVersionSchema.parse( requestedPhpVersion );
+	} catch ( error ) {
+		throw new LoggerError( sprintf( __( 'Unsupported PHP version: %s' ), requestedPhpVersion ) );
+	}
+
+	// If there's already a running Playground instance for this site AND we're not requesting
+	// a different PHP version, pass the command to it…
+	const useCustomPhpVersion = options.phpVersion && options.phpVersion !== site.phpVersion;
 
-		if ( await isServerRunning( site.id ) ) {
-			const result = await sendWpCliCommand( site.id, args );
-			process.stdout.write( result.stdout );
-			process.stderr.write( result.stderr );
-			process.exit( result.exitCode );
+	if ( ! useCustomPhpVersion ) {
+		try {
+			await connect();
+
+			if ( await isServerRunning( site.id ) ) {
+				const result = await sendWpCliCommand( site.id, args );
+				process.stdout.write( result.stdout );
+				process.stderr.write( result.stderr );
+				logger.reportKeyValuePair( 'stdout', result.stdout );
+				logger.reportKeyValuePair( 'stderr', result.stderr );
+				logger.reportKeyValuePair( 'exitCode', String( result.exitCode ) );
+				process.exit( result.exitCode );
+			}
+		} finally {
+			disconnect();
 		}
-	} finally {
-		disconnect();
 	}
 
-	// …If not, instantiate a new Playground instance in the main process
-	const phpVersion = validatePhpVersion( site.phpVersion );
+	// …If not, instantiate a new Playground instance
 	const [ response, closeWpCliServer ] = await runWpCliCommand(
 		siteFolder,
 		phpVersion,
 		site.port,
 		args
 	);
 
-	await response.stderr.pipeTo(
-		new WritableStream( {
-			write( chunk ) {
-				process.stderr.write( chunk );
-			},
-		} )
-	);
+	const stdout = await response.stdoutText;
+	const stderr = await response.stderrText;
+	const exitCode = await response.exitCode;
 
-	await response.stdout.pipeTo(
-		new WritableStream( {
-			write( chunk ) {
-				process.stdout.write( chunk );
-			},
-		} )
-	);
+	process.stdout.write( stdout );
+	process.stderr.write( stderr );
+	logger.reportKeyValuePair( 'stdout', stdout );
+	logger.reportKeyValuePair( 'stderr', stderr );
+	logger.reportKeyValuePair( 'exitCode', String( exitCode ) );
 
 	await closeWpCliServer();
-	process.exit( await response.exitCode );
+	process.exit( exitCode );
 }
 
-function removePathArgumentFromArgv( argv: string[] ) {
+export function removeArgumentFromArgv( argv: string[], argName: string ): string[] {
```

Comment:
```suggestion
function removeArgumentFromArgv( argv: string[], argName: string ): string[] {
```
No need to export, AFAICT